src/main/scala/edu/ie3/simona/agent/participant/hp/HpAgentFundamentals.scala

```diff
@@ -456,6 +457,7 @@ trait HpAgentFundamentals
         s"Unable to initialize heat pump agent '${inputModel.electricalInputModel.getUuid}' without thermal grid model."
       )
     case WithHeatInputContainer(_, thermalGrid) =>
+      ValidationUtils.check(thermalGrid)
```

Comment:
Is there a reason, for doing the validation here? For the `JointGridContainer` we do the check right after reading it (see: `GridProvider`).

Reply:
Right thats the place where we check normally everything. At the moment i don't see a (simple) way to access the complete thermalGrid there. But maybe I oversaw something? If you have a simple approach checking this at GridProvider, feel free. 

Nonetheless maybe a comment in GridProvider would be good, that the validation of thermalGrid is done in HpAgentFundamentals...

Reply:
You could do it in the map method, where we map the thermal grid to its bus. You could first valudate the grid and than map it.

Another option would be to add one of the following lines before the mapping: 
- `.map(ValidationUtils.check)`
- `.map(ValidationUtils.check(_))`